pkg/querier/querier.go

```diff
@@ -109,6 +111,9 @@ func (q *Querier) forGivenIngesters(replicationSet ring.ReplicationSet, f func(l
 
 // Query does the heavy lifting for an actual query.
 func (q *Querier) Query(ctx context.Context, req *logproto.QueryRequest) (*logproto.QueryResponse, error) {
+	// Enforce the query timeout while querying backends
+	ctx, _ = context.WithDeadline(ctx, time.Now().Add(q.cfg.QueryTimeout))
```

Comment:
`govet` is probably right here:
```
pkg/querier/querier.go:115:7: lostcancel: the cancel function returned by context.WithDeadline should be called, not discarded, to avoid a context leak (govet)
	ctx, _ = context.WithDeadline(ctx, time.Now().Add(q.cfg.QueryTimeout))
```
Same applies for all other occurences

Reply:
Thanks for pointing out. I believe you're right. I will deeper investigate it and work on a fix 🙏